### PR TITLE
feature(expo-build-properties): add `ios.ccacheEnabled` option to enable c++ compiler cache

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Closes #28398

This is a re-implementation of #28398, but implemented through `expo-build-properties`.

# How

- [x] Added `ios.ccacheEnabled` to `expo-build-properties`, to add the `apple.ccacheEnabled` podfile property.
- [x] Updated the template to enable `:ccache_enabled` once podfile property is true.
- [x] Update the docs to document this new property.

# Test Plan

See added test

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by gabrieldonadel — APPROVED** [review]

File: templates/expo-template-bare-minimum/ios/Podfile [hunk 1]
```diff
@@ -31,7 +31,8 @@ target 'HelloWorld' do
     react_native_post_install(
       installer,
       config[:reactNativePath],
-      :mac_catalyst_enabled => false
+      :mac_catalyst_enabled => false,
+      :ccache_enabled => !!podfile_properties['apple.ccacheEnabled']
```
Comment (byCedric) [inline templates/expo-template-bare-minimum/ios/Podfile:35]: Oh wait, we convert all booleans to string when writing the JSON. So this needs to be this, right? (can be omitted, `false`, or `true`)

```suggestion
      :ccache_enabled => podfile_properties['apple.ccacheEnabled'] == 'true'
```

**Review by Kudo — APPROVED** [review]
much thanks to @byCedric for helping the work.
just leaving some nit comments here and then good to merge

File: templates/expo-template-bare-minimum/ios/Podfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (byCedric) [reply templates/expo-template-bare-minimum/ios/Podfile:35]: Ah yeah, I figured that out too (see above comment)

**Review by byCedric — COMMENTED** [review]
whoops

File: templates/expo-template-bare-minimum/ios/Podfile (hunk 1 above)
Comment (Kudo) [reply templates/expo-template-bare-minimum/ios/Podfile:35]: i didn't implement other generic types than string: https://github.com/expo/expo/blob/87efd0c9e667151ba26ca047a195d57ea9365273/packages/%40expo/config-plugins/src/utils/BuildProperties.types.ts#L28

the other reason is that on android, gradle.properties only supports strings.

**Pull request merged**